### PR TITLE
Return unclipped gradient from `clip_grad_norm_`

### PR DESCRIPTION
**Pull request by samuelstevens**
Returns the unclipped gradient from `clip_grad_norm_` like [`torch.nn.utils.clip_grad_norm_`](https://pytorch.org/docs/stable/_modules/torch/nn/utils/clip_grad.html#clip_grad_norm_) does.

Initially brought up in #641.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Comment by lzy37ld** [comment]
@sgugger @samuelstevens 
Hi Thanks for this PR. I want to use clip_grad_norm_ in deepspeed intergration. But I notice that we only have a parameter for gradient clipping when use `accelerate launch`. How could I set gradient norm?
